README.md

```diff
@@ -13,19 +13,46 @@ Implement the tool `st` that does exact pattern matching using a suffix tree. Te
 
 # Report
 
-## Specify if you have used a linear time or quadratic time algorithm.
+## Implementation
+We tried implemented both the naive quadratic time algorithm and McCreight’s linear time algorithm. But when timing our implementation of McCreight’s algorithm, we realised we forgot a crucial edge case, which made it run as slow as the naive. We did not fix it before the deadline. 
 
-## Insights you may have had while implementing and comparing the algorithms.
+## Insights
+Indecies are again your enemy. 
+
+Pretty printing is very much needed for nested classes like trees. 
+
+Using the verbose argument is nice, since it makes it easy to switch between whether to print or not. 
 
 ## Problems encountered if any.
 
+For the naive, we had problems with splitting edges/nodes. This was solved by pen and paper, going sytematicly through cases, and using pretty printing. 
+
+As mentioned above, we missed an edge case for McCreight’s algorithm, which resulted in it being half-way done in this version of the handin. 
+
 ## Correctness
 
-*Describe experiments that verifies the correctness of your implementations.*
+### Building the tree
+We started by using small examples, which could be done by hand and verified. 
+
+We then used random examples and tested whether the output of the algorithms gave a 'valid' tree. A 'valid' tree might not be the correct one, but it contains all the expected leaves.
+
+### Search 
+
+Likewise we first used some known examples for verifying the implementation of search. 
+
+We then compared the results of the search algorithm on the trees for random strings with the results from the kmp implementation from the last handin. 
+
+### Conclusion of correctness
+
+With these different methods of testing the contruction of trees and search, we are confident, the trees construction for the naive algorithm and search are correct. 
 
 ## Running time
 
-*Describe experiments that verifies that your implementation of `st` uses no more time than O(n) or O(n²) (depending on the algorithm) for constructing the suffix tree and no more than O(m) for searching for a given read in it. Remember to explain your choice of test data. What are “best” and “worst” case inputs?*
+We can see on plots, that it is O(n²). 
```

Comment:
Well, I can't. You haven't given me any plots. (A link to a spreadsheet is not a plot).

You don't have to now, I trust you, but I hope those plots are of the observed divided by expected running time, 'cause I'm getting tired of telling people that I cannot look at some random plot and determine that it is $O(n^2)$ and not for example $O(n^{2.3})$.